### PR TITLE
Upgrade rust driver

### PR DESCRIPTION
**Pull request by adespawn**
Update rust driver to version 0.15.0. 

Move out of legacy API, while still converting data to Row type

### CODE REVIEW

File: src/result.rs [hunk 1]
```diff
@@ -69,64 +73,83 @@ pub enum CqlType {
...
-        internal: QueryResult,
-    ) -> Result<QueryResultWrapper, IntoLegacyQueryResultError> {
-        Ok(QueryResultWrapper {
-            internal: internal.into_legacy_result()?,
-        })
+    pub fn from_query(internal: QueryResult) -> QueryResultWrapper {
+        let value = match internal.clone().into_rows_result() {
+            Ok(v) => QueryResultVariant::RowsResult(v),
+            Err(_) => QueryResultVariant::EmptyResult(internal),
+        };
```
Comment (wprzytula) [inline src/result.rs:80]: This is not about the number of rows returned, but about the kind of the Result response: Rows or not Rows. For non-Rows Result kinds, this conversion fails.

File: src/result.rs [hunk 2]
```diff
@@ -69,64 +73,83 @@ pub enum CqlType {
...
+            QueryResultVariant::RowsResult(v) => v,
+            QueryResultVariant::EmptyResult(_) => {
                 return None;
             }
         };
 
+        let rows = r2
+            .rows::<Row>()
+            .expect("Expected result to bo converted into Row")
+            .flatten();
```
Comment (wprzytula) [inline src/result.rs:96]: This is absolutely unacceptable!!! We mustn't panic on bad data received!

File: src/result.rs [hunk 3]
```diff
@@ -69,64 +73,83 @@ pub enum CqlType {
...
         Some(
-            rows.iter()
-                .map(|f| RowWrapper {
-                    internal: f.columns.clone(),
-                })
-                .collect(),
+            rows.map(|f| RowWrapper {
+                internal: f.columns.clone(),
```
Comment (wprzytula) [inline src/result.rs:99]: Why clone the columns instead of moving them out?

**Review by wprzytula — CHANGES_REQUESTED** [review]

File: src/result.rs [hunk 4]
```diff
@@ -70,59 +78,79 @@ pub enum CqlType {
...
+        QueryResultWrapper { internal: value }
     }
 
     #[napi]
-    pub fn get_rows(&self) -> Option<Vec<RowWrapper>> {
-        let rows = match &self.internal.rows {
-            Some(r) => r,
-            None => {
-                return None;
+    pub fn get_rows(&self) -> napi::Result<Option<Vec<RowWrapper>>> {
```
Comment (wprzytula) [inline src/result.rs:89]: So you changed this function's signature and didn't update any calling code. Is this function currently unused?

Comment (adespawn) [reply src/result.rs:89]: It's used from nodeJS

**Review by wprzytula — APPROVED** [review]

**Pull request merged**